lib/route.js

```diff
@@ -297,8 +298,9 @@ function buildRouting (options) {
       if (ip !== undefined && req.headers['x-forwarded-host']) {
         hostname = req.headers['x-forwarded-host']
       }
+      proto = req.headers['x-forwarded-proto'] || proto
     }
-
+    var protocol = proto.split(/\s*,\s*/)[0]
```

Comment:
regex can be very slow. Can you explain what this is doing and if there is possibly a better way to do it?

Reply:
Is the regexp needed only if there is the x-forwarded-proto header?